src/abstractdataframe/abstractdataframe.jl

```diff
@@ -405,31 +412,109 @@ describe(df)
     end
-end
-
-function StatsBase.describe(io::IO, X::AbstractVector{Union{T, Missing}}) where T
-    missingcount = count(ismissing, X)
-    pmissing = 100 * missingcount/length(X)
-    if pmissing != 100 && T <: Real
-        show(io, StatsBase.summarystats(collect(skipmissing(X))))
-    else
-        println(io, "Summary Stats:")
+    # Define 4 functions for getting summary statistics 
+    # use a dict because we dont know which measures the user wants
+    function get_stats(col::AbstractArray{T} where T <: Real)
+        stats = summarystats(col)
+        stats_dict = Dict(
+            :mean => stats.mean,
+            :sd => sqrt(var(col)),
+            :min => stats.min ,
+            :q25 => stats.q25,
+            :median => stats.median,
+            :q75 => stats.q75,
+            :max => stats.max,
+            :datatype=> eltype(col),
+            :Nmissing => nothing,
+            :Nunique => nothing
+        )
+    end
+    
+    function get_stats(col::AbstractArray{Union{T, Missing}} where T <: Real)
+        stats = summarystats(collect(skipmissing(col)))
+        stats_dict = Dict(
+            :mean => stats.mean,
+            :sd => sqrt(var(collect(skipmissing(col)))),
+            :min => stats.min ,
+            :q25 => stats.q25,
+            :median => stats.median,
+            :q75 => stats.q75,
+            :max => stats.max,
+            :datatype=> Missings.T(eltype(col)),
+            :Nmissing => count(ismissing(col)),
+            :Nunique => nothing
+        )
     end
-    println(io, "Length:         $(length(X))")
-    println(io, "Type:           $(eltype(X))")
-    !(T <: Real) && println(io, "Number Unique:  $(length(unique(X)))")
-    println(io, "Number Missing: $(missingcount)")
-    @printf(io, "%% Missing:      %.6f\n", pmissing)
-    return
+    
+    function get_stats(col)
+        stats_dict = Dict(
+            :mean => nothing,
+            :sd => nothing,
+            :min => nothing,
+            :q25 => nothing,
+            :median => nothing,
+            :q75 => nothing,
+            :max => nothing,
+            :datatype=> eltype(col),
+            :Nmissing => nothing,
+            :Nunique => length(unique(col))
+        )   
+    end
+    
+    function get_stats(col:: AbstractArray{Union{T, Missing}} where T <: Any)
+        stats_dict = Dict(
+            :mean => nothing,
+            :sd => nothing,
+            :min => nothing,
+            :q25 => nothing,
+            :median => nothing,
+            :q75 => nothing,
+            :max => nothing,
+            :datatype=> Missings.T(eltype(col)),
+            :Nmissing => count(ismissing(col)),
+            :Nunique => length(unique(col))
+        )    
+    end 
+    # Takes in a column and returns a row vector of the statistics
+    function spread(col)
+        d = get_stats(col) # a Dict
+        row = [d[f] for f in colstats]
+        reshape(row, 1, length(row))
+    end
+    # Append the summary stats into a matrix
```

Comment:
Creating a matrix is wasteful since what we need in the end is a `DataFrame`. So better collect values directly into a `DataFrame`.

Reply:
Thanks for all the feedback. I'm working on changing the code now. Do you have any idea what the best way to do this would be? 

```
df = DataFrame(rand(10,10))
df[1,:] = rand(10)' # normal vector doesn't work either

>MethodError: Cannot `convert` an object of type Array{Float64,1} to an object of type Float64
```

It looks like you can replace a row in a DataFrame with a number, which then propagates across the whole row, but you can't replace it with a row. The expected behavior was that of a `Matrix`, where you can do

```
x = rand(10,10)
x[1,:] = ones(10) # no transpose needed
```